src/ripple/protocol/impl/LedgerFormats.cpp

```diff
@@ -165,4 +170,49 @@ LedgerFormats::getInstance ()
     return instance;
 }
 
+std::pair<RPC::Status, LedgerEntryType>
+chooseLedgerEntryType(Json::Value const& params)
+{
+    std::pair<RPC::Status, LedgerEntryType> result{RPC::Status::OK, ltINVALID};
+    if (params.isMember(jss::type))
+    {
+        static
+        std::array<std::pair<char const *, LedgerEntryType>, 9> const types
+            {{
+                { jss::account,     ltACCOUNT_ROOT  },
+                { jss::amendments,  ltAMENDMENTS    },
+                { jss::directory,   ltDIR_NODE      },
+                { jss::fee,         ltFEE_SETTINGS  },
+                { jss::hashes,      ltLEDGER_HASHES },
+                { jss::offer,       ltOFFER         },
+                { jss::signer_list, ltSIGNER_LIST   },
+                { jss::state,       ltRIPPLE_STATE  },
+                { jss::ticket,      ltTICKET        }
+            }};
+
+        auto const& p = params[jss::type];
+        if (!p.isString())
+        {
+            result.first = RPC::Status{rpcINVALID_PARAMS,
+                                       "Invalid field 'type', not string."};
+            assert(result.first.type() == RPC::Status::Type::error_code_i);
+            return result;
+        }
+
+        auto const filter = p.asString ();
+        auto iter = std::find_if (types.begin (), types.end (),
+            [&filter](decltype (types.front ())& t)
+                { return t.first == filter; });
+        if (iter == types.end ())
+        {
+            result.first = RPC::Status{rpcINVALID_PARAMS,
+                                       "Invalid field 'type'."};
```

Comment:
Adding a more specific cause may be helpful to the client. I suggest "Invalid field 'type', unknown value." or "Invalid field 'type', invalid value."